soc/wch/ch32v00x/Kconfig.defconfig

```diff
@@ -0,0 +1,24 @@
+# Copyright (c) 2024 Michael Hope
+# SPDX-License-Identifier: Apache-2.0
+
+if SOC_CH32V003
+
+config NUM_IRQS
+	default 48
+
+config SYS_CLOCK_HW_CYCLES_PER_SEC
+	default 48000000
```

Comment:
can be obtained from DT (see other socs)

Reply:
I tried the following patch:

```diff
diff --git a/dts/riscv/wch/ch32v00x.dtsi b/dts/riscv/wch/ch32v00x.dtsi
index 0f611324e18..0066dc1992d 100644
--- a/dts/riscv/wch/ch32v00x.dtsi
+++ b/dts/riscv/wch/ch32v00x.dtsi
@@ -22,6 +22,7 @@
                        device_type = "cpu";
                        compatible = "wch,qingke-v2";
                        reg = <0>;
+                       clock-frequency = <DT_FREQ_M(48)>;
                };
        };
 
diff --git a/soc/wch/ch32v00x/Kconfig.defconfig b/soc/wch/ch32v00x/Kconfig.defconfig
index 34439fa61b8..3cc679acba3 100644
--- a/soc/wch/ch32v00x/Kconfig.defconfig
+++ b/soc/wch/ch32v00x/Kconfig.defconfig
@@ -7,6 +7,7 @@ config NUM_IRQS
        default 48
 
 config SYS_CLOCK_HW_CYCLES_PER_SEC
+       default $(dt_node_int_prop_int,/cpus/cpu@0,clock-frequency)
        default 48000000
```

It results in the following error:

```
zephyr/sys_clock.h:151:2: error: #error "SYS_CLOCK_HW_CYCLES_PER_SEC must be non-zero!"
```

```
$ cat build/zephyr/zephyr.dts
/dts-v1/;

/ {
	#address-cells = < 0x1 >;
	#size-cells = < 0x1 >;
	model = "ch32v003evt";
	compatible = "wch,ch32v003";
	cpus {
		#address-cells = < 0x1 >;
		#size-cells = < 0x0 >;
		cpu0: cpu@0 {
			device_type = "cpu";
			compatible = "wch,qingke-v2";
			reg = < 0x0 >;
			clock-frequency = < 0x2dc6c00 >;
...
```

Reply:
does `wch,qingke-v2` binding include `cpu.yml`?

Reply:
Thank you - the desired change is in place now.